docs/content/quickstart/parameters.md

```diff
@@ -0,0 +1,147 @@
+
+Parameters:
+Name   Description                           Type     Default   Required   Applies To
+name   Name of to whom we should say hello   string   llama     false      All Actions
+
+No outputs defined
+
+No custom actions defined
+
+No dependencies defined
+```
+
+The output tells us that the bundle has one optional string parameter, name, which defaults to "llama".
+The name parameter applies to "All Actions" meaning that this parameter can be specified with every action that the bundle defines.
+Since it says that no custom actions are defined, the bundle only supports the built-in actions of install, upgrade, and uninstall.
+
+## Use the default parameter values
+
+
+First install the bundle and do not specify any parameters so that you can observe how the bundle installs without any customization.
+
+```console
+$ porter install hello-llama --reference getporter/hello-llama:v0.1.1
+installing hello-llama...
+executing install action from hello-llama (installation: hello-llama)
+Hello, llama
+execution completed successfully!
+```
+
+The bundle printed "Hello, llama" using the default value for the name parameter.
+
+## Specify a parameter with a flag
+Next upgrade the installation and change the name parameter to another value.
+Parameters are specified with the \--param flag: 
+
+```console
+$ porter upgrade hello-llama --param name=Porter
+upgrading hello-llama...
+executing upgrade action from hello-llama (installation: hello-llama)
+Porter 2.0
+execution completed successfully!
+```
+
+## Create a Parameter Set
+Setting a parameter value individually on the command line with the \--param flag works well when trying out a bundle.
+When working with a bundle often though, remembering and typing out every parameter values every time is error prone.
+Parameter Sets store a set of parameter values to use with a bundle.
+Even more powerful, the value of the parameters can come from the following sources:
+
+* hard-coded value
+* environment variable
+* file
+* command output
+* secret
+
+Some parameters may be sensitive, for example a database connection string or oauth token.
+For improved security, and to limit exposure of sensitive values, it is recommended that you source those parameter values from a secret store such as Hashicorp Vault or Azure Key Vault.
+
+Create a parameter set for the hello-llama with the `porter parameters generate` command. It is an interactive command that walks through setting values for every parameter in the specified bundle.
+
+```console
+$ porter parameters generate hello-llama --reference getporter/hello-llama:v0.1.1
+Generating new parameter set hello-llama from bundle hello-llama
+==> 2 parameters declared for bundle hello-llama
+
+? How would you like to set parameter "name"
+   [Use arrows to move, space to select, type to filter]
+  secret
+> specific value
+  environment variable
+  file path
+  shell command
+
+? Enter the value that will be used to set parameter "name"
+  Porter
+```
+
```

Comment:
What if I want to accept a default value for a particular parameter?

Reply:
Can there only be one parameter set per bundle? What happens if I re-run generate?

Reply:
> What if I want to accept a default value for a particular parameter?

Let me make an issue for that. Right now the answer is manually edit the parameter set to remove the parameter mapping from the file.

Reply:
> Can there only be one parameter set per bundle? What happens if I re-run generate?

I have answers but I'm not sure if that level of detail belongs in the quickstart, which is a task focused document and isn't intended to be a complete reference of the topic. A better place to answer those questions is porter.sh/parameters I think. We already say in the command-line documentation for the --parameter-set flag that it can be used more than once.

1. Yes you can define multiple parameter sets and use them when installing a bundle.
2. Re-running generate, runs you through the same process as the first time. If you choose the same parameter set name the previous parameter set is overwritten.

Reply:
Cool, wasn't saying that these things need to defined here but wondering if folks have more questions if they were answered somewhere.